lib/linter.js

```diff
@@ -135,36 +139,41 @@ function lint(objectName, object, options = {}) {
                 });
             }
             if (rule.pattern) {
+                const { omit, property, split, value } = rule.pattern;
+                const target = object[property]
+
                 let components = [];
-                if (rule.pattern.split) {
-                    components = object[rule.pattern.property].split(rule.pattern.split);
+                if (split) {
+                    components = target.split(split);
                 }
                 else {
-                    components.push(object[rule.pattern.property]);
+                    components.push(target);
                 }
-                const re = new RegExp(rule.pattern.value);
+                const re = new RegExp(value);
                 for (let component of components) {
-                    if (rule.pattern.omit) component = component.split(rule.pattern.omit).join('');
+                    if (omit) component = component.split(omit).join('');
                     if (component) {
                         ensure(rule, () => {
-                            should(re.test(component)).be.exactly(true,rule.description);
+                            should(re.test(component)).be.exactly(true, rule.description);
                         });
                     }
                 }
             }
             if (rule.notContain) {
-                for (const property of rule.notContain.properties) {
+                const { value, properties } = rule.notContain;
+                for (const property of properties) {
                     if (object[property] && (typeof object[property] === 'string') &&
-                        (object[property].indexOf(rule.notContain.value)>=0)) {
+                        (object[property].indexOf(value) >= 0)) {
```

Comment:
Should these be `target` from above?

Reply:
Weirdly target is just for `if (rule.pattern) {`. I was simplifying that one piece of code, and trying to avoid a megarefactor of the whole thing. I'll go further another time, but some rules have property and some have properties, meaning we cant just do a simple `target = X` at the top.